packages/ui/src/views/repo/components/branch-selector/branch-selector.tsx

```diff
@@ -0,0 +1,55 @@
+import { Button, DropdownMenu, DropdownMenuTrigger, Icon, Text } from '@/components'
+import { cn } from '@utils/cn'
+
+import { BranchSelectorDropdown, type BranchSelectorDropdownProps } from './branch-selector-dropdown'
+
+interface BranchSelectorProps extends BranchSelectorDropdownProps {
+  size?: 'default' | 'sm'
+  prefix?: string
+  className?: string
+}
+
+export const BranchSelector: React.FC<BranchSelectorProps> = ({
+  name,
+  branchList,
+  tagList,
+  size = 'default',
+  selectBranch,
+  prefix,
+  className,
+  repoId,
+  spaceId
+}) => {
+  const isTag = tagList.some(tag => tag.name === name) ?? false
```

Comment:
`some` returns a boolean anyway, so you don't need to default to `false`
```suggestion
  const isTag = tagList.some(tag => tag.name === name)
```